dbt/adapters/databricks/credentials.py

```diff
@@ -241,172 +250,174 @@ def extract_cluster_id(cls, http_path: str) -> Optional[str]:
-                redirect_url=redirect_url,
-                scopes=scopes,
-            )
-            # optional branch. Try and keep going if it does not work
-            try:
-                # try to get cached credentials
-                credsdict = self.get_sharded_password("dbt-databricks", host)
-
-                if credsdict:
-                    provider = SessionCredentials.from_dict(oauth_client, json.loads(credsdict))
-                    # if refresh token is expired, this will throw
-                    try:
-                        if provider.token().valid:
-                            self._credentials_provider = provider.as_dict()
-                            if json.loads(credsdict) != provider.as_dict():
-                                # if the provider dict has changed, most likely because of a token
-                                # refresh, save it for further use
-                                self.set_sharded_password(
-                                    "dbt-databricks", host, json.dumps(self._credentials_provider)
-                                )
-                            return provider
-                    except Exception as e:
-                        # SPA token are supposed to expire after 24h, no need to warn
-                        if SPA_CLIENT_FIXED_TIME_LIMIT_ERROR in str(e):
-                            logger.debug(CredentialLoadError(e))
-                        else:
-                            logger.warning(CredentialLoadError(e))
-                        # whatever it is, get rid of the cache
-                        self.delete_sharded_password("dbt-databricks", host)
-
-            # error with keyring. Maybe machine has no password persistency
-            except Exception as e:
-                logger.warning(CredentialLoadError(e))
-
-            # no token, go fetch one
-            consent = oauth_client.initiate_consent()
-
-            provider = consent.launch_external_browser()
-            # save for later
-            self._credentials_provider = provider.as_dict()
-            try:
-                self.set_sharded_password(
-                    "dbt-databricks", host, json.dumps(self._credentials_provider)
-                )
-            # error with keyring. Maybe machine has no password persistency
-            except Exception as e:
-                logger.warning(CredentialSaveError(e))
+        assert self._credentials_manager is not None, "Credentials manager is not set."
+        return self._credentials_manager
 
-            return provider
 
-        finally:
-            self._lock.release()
+class BearerAuth(AuthBase):
+    """This mix-in is passed to our requests Session to explicitly
+    use the bearer authentication method.
 
-    def set_sharded_password(self, service_name: str, username: str, password: str) -> None:
-        max_size = MAX_NT_PASSWORD_SIZE
+    Without this, a local .netrc file in the user's home directory
```

Comment:
Is it safe to remove it seems databricks cli will not use .netrc anymore? https://docs.databricks.com/en/dev-tools/cli/profiles.html @benc-db @jackyhu-db

Reply:
I think `.netrc` is used by python `requests` which might be used by databricks-sdk code as well?

Reply:
yeah, the .netrc thing bit us in the past.